### PR TITLE
Error implementation

### LINKED ISSUE #64: [APP - RUST] - Simplify Error management in commands
We can avoid converting each error to String in commands using thiserror from macro and deriving Serialize for the newer Error enum. 
See [this discussion](https://github.com/tauri-apps/tauri/discussions/3913) for the implementation details.
It may be useful to introduce logging to see from which error, the newer one are derived using fmt::Debug

### PR DESCRIPTION
**Pull request by newfla**
Closes #64.

I'm not sure about the name FORCE_DEBUG_LOG_FLAG and forwarding logs to js console in debug builds. 
What do you suggest we do? @brainrepo

### CODE REVIEW

**Comment by brainrepo** [comment]
I have a doubt, maybe coming from my rust limited knowledge, are you preserving all the error messages. For example I can't see "error preserving bucket connection" I imagine that this error is handled somewhere but I can't see where. 
Are we distingue the specific error generated from an s3 flow? Are we returning back the internal error info?

**Comment by brainrepo** [comment]
LGTM! 
Just a little question.

File: packages/app/src-tauri/src/utils/log.rs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+pub fn log_status() -> bool {
+    LOG_ENABLED.load(Ordering::Relaxed)
+}
+
+fn filter_log(_: &Metadata<'_>) -> bool {
+    log_status()
+}
+
+#[macro_export]
+macro_rules! log_if_error {
```
Comment (brainrepo) [inline packages/app/src-tauri/src/utils/log.rs:36]: maybe a more explicit name cold be helpful, something like `log_if_error_or_return`

File: packages/app/src-tauri/src/commands/ftp.rs [hunk 2]
```diff
@@ -179,9 +170,7 @@ mod test {
 
         // Hopefully the server is up =D
         sleep(Duration::from_secs(2)).await;
-        assert!(ftp_upload(payload)
-            .await
-            .is_err_and(|err| err == "Failed to connect to server"));
```
Comment (brainrepo) [inline packages/app/src-tauri/src/commands/ftp.rs:184]: How can we track the specific errors like:
- Failed to connect to server
- Failed to login to server
- Failed to open file

**Comment by newfla** [comment]
> @newfla I have a doubt, maybe coming from my rust limited knowledge, are you preserving all the error messages. For example I can't see "error preserving bucket connection" I imagine that this error is handled somewhere but I can't see where. Are we distingue the specific error generated from an s3 flow? Are we returning back the internal error info?

All `s3::error::S3Error` are going to be translated to the frontend as "Error while operationing on s3 storage". `log_if_error` logs errors using `:? fmt::Debug ` that shows also the original error, e.g networ.rs.:

`[2023-11-26][09:03:52][ERROR][fourviere_podcast::commands::network] [fetch_feed] error: Network(reqwest::Error { kind: Request, url: Url { scheme: "https", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("notaserver.it")), port: None, path: "/", query: None, fragment: None }, source: hyper::Error(Connect, ConnectError("dns error", Custom { kind: Uncategorized, error: "failed to lookup address information: Name or service not known" })) })`

What do you think? May be okay?

File: packages/app/src-tauri/src/utils/log.rs (hunk 1 above)
Comment (newfla) [reply packages/app/src-tauri/src/utils/log.rs]: Thx, renamed as `log_if_error_and_return`

**Pull request merged**